Refactor SlackNotificationPayloadContent - extract delegate class

A4Vision opened the pull request:
Extract class for the commits list in SlackNotificationPayloadContent.

A4Vision commented:
Hello @marvin-w , would you like to review this PR ?

marvin-w commented on tcslackbuildnotifier-core/src/main/java/slacknotifications/teamcity/payload/content/SlackNotificationPayloadContent.java:
```diff
@@ -99,11 +96,10 @@ public SlackNotificationPayloadContent(SBuildServer server, SBuildType buildType
             populateCommonContent(server, sRunningBuild, previousBuild, buildState);
     		populateMessageAndText(sRunningBuild, buildState);
-            populateCommits(sRunningBuild);
+			payloadCommits.populateCommits(sRunningBuild);
```
Hey, please fix the identation in this file.

marvin-w commented on tcslackbuildnotifier-core/src/main/java/slacknotifications/teamcity/payload/content/SlackNotificationPayloadContent.java:
```diff
@@ -77,7 +73,7 @@
 
     public SlackNotificationPayloadContent(){
-
+		payloadCommits = new PayloadContentCommits();
```
you could move this declaration to the variable initializer at the top of the file and save the additional 3 declarations.

You could also use this() to call the next constructor, which would also be a better practice. (I know this is not the case right now, but since the title is called refactoring it might make sense ;) )

marvin-w commented on tcslackbuildnotifier-core/src/main/java/slacknotifications/teamcity/payload/content/PayloadContentCommits.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.List;
+
+public class PayloadContentCommits {
+    List<Commit> commits;
```
make it private please

marvin-w commented on tcslackbuildnotifier-core/src/main/java/slacknotifications/teamcity/payload/content/PayloadContentCommits.java:
```diff
@@ -0,0 +1,45 @@
+    public void setCommits(List<Commit> commits) {
+        this.commits = commits;
+    }
+}
```
add the missing newline please

marvin-w commented on tcslackbuildnotifier-core/src/main/java/slacknotifications/teamcity/payload/content/PayloadContentCommits.java:
```diff
@@ -0,0 +1,45 @@
+        commits = new ArrayList<Commit>();
+    }
+
+    void populateCommits(SRunningBuild sRunningBuild) {
```
make this method public since thats the only real thing this class does. (which is good)

A4Vision commented:
Hi @marvin-w , thank you for the review !
* Indentation - reformatted file with 4 spaces tab
* Initialization - initialized with a default initializer.
* Changed the field `commits` to private
* Added missing new line at the end of file.
* Made the new function public.

Are you sure you want to make `populateCommits` public ? Package is enough here.

Best,
 Assaf
